src/test/scala/org/broadinstitute/dsde/agora/server/AgoraTestSuite.scala

```diff
@@ -0,0 +1,27 @@
+package org.broadinstitute.dsde.agora.server
+
+import com.github.simplyscala.{MongoEmbedDatabase, MongodProps}
+import org.broadinstitute.dsde.agora.server.dataaccess.AgoraDao
+import org.broadinstitute.dsde.agora.server.dataaccess.mongo.MethodsDbTest
+import org.broadinstitute.dsde.agora.server.webservice.ApiServiceSpec
+import org.scalatest.{BeforeAndAfterAll, Suites}
+
+trait AgoraDbTest {
+  val agoraDao = AgoraDao.createAgoraDao
+}
+
+class AgoraTestSuite extends Suites(new ApiServiceSpec, new MethodsDbTest) with BeforeAndAfterAll with MongoEmbedDatabase {
+  var mongoProps: MongodProps = null
+
+  override def beforeAll() {
+    println("Starting embedded mongo db instance.")
+    mongoProps = mongoStart(port = Agora.server.mongoDbPort)
+    println("Starting Agora web services.")
+    Agora.start()
+  }
+
+  override def afterAll() {
+    println("Stopping embedded mongo db instance.")
+    mongoStop(mongoProps)
```

Comment:
Agora has a shutdown hook for stopping so it was a bit redundant. We can add it back though if you like the explicit stop call. You get duplicate stop messages in the logs though which is a bit odd.